atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -429,20 +429,29 @@ <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionRea
     void registerClosingCallback(Runnable closingCallback);
 
     /**
-     * This method can be used for direct control over the lifecycle of a batch of transactions. For example, if the
-     * work done in each given transaction is interactive and cannot be expressed as a {@link TransactionTask} ahead of
-     * time, this method allows for a long lived transaction object. For any data read or written to the transaction to
-     * be valid, the transaction must be committed by calling {@link OpenTransaction#finish(TransactionTask)} to
-     * also perform additional cleanup. Note that this does not clean up the pre commit condition associated with that
-     * task. The order of transactions returned corresponds with the pre commit conditions passed in, however there are
+     * This method can be used for direct control over the lifecycle of a batch of transactions.
+     * For example, if the work done in each given transaction is interactive and cannot be expressed as a
+     * {@link TransactionTask} ahead of time, this method allows for a long lived transaction object.
+     * <p>
+     * For any data read or written to the transaction to be valid, the transaction must be committed explicitly
+     * by the caller with {@link Transaction#commit()}.
+     * <p>
+     * {@link PreCommitCondition#cleanup()} associated with a transaction will be run when the respective transaction
+     * commits or aborts.
+     * All pre-commit conditions are also cleaned in case {@link #startTransactions(List)} throws.
+     * <p>
+     * Note the caller must call {@link OpenTransaction#close()} after the transaction is committed to perform
+     * additional cleanup. Failure to do so might incur in
+     * <p>
+     * The order of transactions returned corresponds with the pre commit conditions passed in, however there are
      * no guarantees on the ordering of transactions returned with respect to their start timestamp.
      *
      * @return a batch of transactions with associated immutable timestamp locks
      * @deprecated Similar functionality will exist, but this method is likely to change in the future
      */
     @Deprecated
     @Timed
-    List<OpenTransaction> startTransactions(List<? extends PreCommitCondition> condition);
+    List<? extends OpenTransaction> startTransactions(List<? extends PreCommitCondition> condition);
```

Comment:
For my edification: what do you get out of doing "? extends ", other than the callers now having to do the same thing?